### PR TITLE
chore(qa): Enable metrics for nginx, uwsgi and fence logs

### PR DESCRIPTION
**Pull request by themarcelor**
Enabling metrics to be scraped by Prometheus.

This change depends on the following PRs:

~1. https://github.com/uc-cdis/cloud-automation/pull/1349.~
2. https://github.com/uc-cdis/fence/pull/810.

Testing:
- [x] A) build a fence branch using the base image from this cloud-automation branch
- [x] B) deploy the fence branch
- [x] C) verify that fence works, and prometheus metrics appear
- [x] D) merge and tag the cloud-automation branch
- [x] E) flip the fence branch over to the tagged base image
- [x] F) merge the fence branch

Verifying metrics' endpoints:
```
qa-dcp@cdistest_dev_admin:~$ ~$ kc exec -it fence-deployment-9bf769-4nvsm -c fence -- curl -L -s -o /dev/null -w "%{http_code}" -X GET http://localhost/metrics
200
qa-dcp@cdistest_dev_admin:~$ kc exec -it fence-deployment-9bf769-4nvsm -c fence -- curl -L -s -o /dev/null -w "%{http_code}" -X GET http://localhost:9117/metrics
200
qa-dcp@cdistest_dev_admin:~$ kc exec -it fence-deployment-9bf769-4nvsm -c fence -- curl -L -s -o /dev/null -w "%{http_code}" -X GET http://localhost:9113/metrics
200
qa-dcp@cdistest_dev_admin:~$ kc exec -it fence-deployment-9bf769-4nvsm -c fence -- curl -L -s -o /dev/null -w "%{http_code}" -X GET http://localhost:4040/metrics
200
qa-dcp@cdistest_dev_admin:~$ kc exec -it fence-deployment-9bf769-4nvsm -c fence -- curl -L -s -o /dev/null -w "%{http_code}" -X GET http://localhost:6567/aggregated_metrics
200
qa-dcp@cdistest_dev_admin:~$
```

Testing backwards compatibility with `nginx-prometheus-exporter-wrapper`:
```
jenkins-blood@vm:~$ kc exec -it fence-deployment-69db9b46c6-phpsm -c nginx-prometheus-exporter-wrapper
error: you must specify at least one command for the container
jenkins-blood@vm:~$ kc exec -it fence-deployment-69db9b46c6-phpsm -c nginx-prometheus-exporter-wrapper -- bash
bash-5.0$ ps fuxa
PID   USER     TIME  COMMAND
    1 1001      0:00 /bin/bash -c nginx_status_response=$(curl -L -s -o
    8 1001      0:00 sleep infinity
    9 1001      0:00 bash
   14 1001      0:00 ps fuxa
```

### CODE REVIEW

File: kube/services/fence/fence-deploy.yaml [hunk 1]
```diff
@@ -46,6 +46,8 @@ spec:
 # DEPRECATED! Remove when all commons are no longer using local_settings.py
 #             for fence.
 # -----------------------------------------------------------------------------
+        - name: logs-folder
```
Comment (frickjack) [inline kube/services/fence/fence-deploy.yaml:49]: Should we add pormetheus annotations to the pod?
   https://github.com/uc-cdis/cloud-automation/blob/master/kube/services/ambassador/ambassador-deploy.yaml#L17

File: kube/services/fence/fence-deploy.yaml [hunk 2]
```diff
@@ -134,6 +157,8 @@ spec:
 # DEPRECATED! Remove when all commons are no longer using local_settings.py
 #             for fence.
 # -----------------------------------------------------------------------------
+          - name: logs-folder
+            mountPath: "/var/log/nginx/"
```
Comment (frickjack) [inline kube/services/fence/fence-deploy.yaml:161]: Are these logs being rotated?

Comment (themarcelor) [reply kube/services/fence/fence-deploy.yaml:161]: Nginx does not provide a mechanism to rotate logs (according to some quick googling).
Do you have any ideas?
How about a cronjob to run some logrotate?

File: kube/services/fence/fence-deploy.yaml (hunk 1 above)
Comment (themarcelor) [reply kube/services/fence/fence-deploy.yaml:49]: Oh wow, this is amazing! Thanks for the suggestion.

File: Docker/python-nginx/python3.6-alpine3.7/dockerrun.sh [hunk 3]
```diff
@@ -99,5 +99,11 @@ fi
   fi
   run uwsgi --ini /etc/uwsgi/uwsgi.ini
 ) &
+(
+  while true; do logrotate --force /etc/logrotate.d/nginx; sleep 86400; done
+) &
+(
+  while true; do curl -s http://127.0.0.1:9117/metrics > /var/www/metrics/metrics.txt && curl -s http://127.0.0.1:9113/metrics >> /var/www/metrics/metrics.txt && curl -s http://127.0.0.1:4040/metrics >> /var/www/metrics/metrics.txt; sleep 10; done
```
Comment (frickjack) [inline Docker/python-nginx/python3.6-alpine3.7/dockerrun.sh:106]: why don't you split this into multiple lines, or move it into a function?

File: gen3/lib/testData/test1.manifest.g3k/expectedFenceResult.yaml [hunk 4]
```diff
@@ -93,6 +95,27 @@ spec:
         # nginx group in current images
         fsGroup: 101
       containers:
+      - name: nginx-prometheus-exporter
```
Comment (frickjack) [inline gen3/lib/testData/test1.manifest.g3k/expectedFenceResult.yaml:98]: you need to have 3 sidecars to convert status endpoints into something prometheus can harvest?
can't we just have one script that does that, and writes out the `metrics.txt` in the fence container?

Comment (themarcelor) [reply gen3/lib/testData/test1.manifest.g3k/expectedFenceResult.yaml:98]: I thought about using existing exporters instead of implementing our own.
But.. yeah, we could implement our own exporters later.

File: gen3/lib/testData/test1.manifest.g3k/expectedFenceResult.yaml [hunk 5]
```diff
@@ -209,6 +218,33 @@ spec:
             echo "${FENCE_PUBLIC_CONFIG:-""}" > "/var/www/fence/fence-config-public.yaml"
             python /var/www/fence/yaml_merge.py /var/www/fence/fence-config-public.yaml /var/www/fence/fence-config-secret.yaml > /var/www/fence/fence-config.yaml
             bash /fence/dockerrun.bash && if [[ -f /dockerrun.sh ]]; then /dockerrun.sh; fi
+      - name: nginx-prometheus-exporter-wrapper
+        image: quay.io/cdis/nginx-prometheus-exporter-wrapper:latest
```
Comment (frickjack) [inline gen3/lib/testData/test1.manifest.g3k/expectedFenceResult.yaml:222]: Provide a manifest image key with a fallback.
Merge the new Dockerfile first, tag the image, then submit a separate PR with the k8s yaml files that reference the tagged image as a fallback
